integration-tests/src/test/java/io/testerra/test/status/CheckTestStatusTest.java

```diff
@@ -0,0 +1,220 @@
+                {"testT02_crashedDataProvider", Status.SKIPPED, "java.lang.AssertionError"},
+                {"testT03_AssertFailedDataProvider", Status.SKIPPED, "java.lang.AssertionError"}
+        };
+    }
+
+    @DataProvider
+    public static Object[][] provideTestMethodsRetried() {
+        return new Object[][]{
+                {"testCaseOne", Status.RECOVERED},
+                {"test_retryOnWebDriverException", Status.RECOVERED},
+                {"test_retryOnUnreachableBrowserException", Status.RECOVERED},
+                {"test_retryOnJsonException", Status.RECOVERED},
+                {"test_RetryFailed", Status.FAILED},
+                {"test_RetryExpectedFailed", Status.FAILED_EXPECTED},
+                {"test_RetryValidExpectedFailed", Status.FAILED_EXPECTED},
+                {"test_RetryInvalidExpectedFailed", Status.FAILED}
+        };
+    }
+
+    @DataProvider
+    public static Object[][] provideTestMethodsNoRetry() {
+        return new Object[][]{
+                {"test_NoRetry"}
+        };
+    }
+
+    @DataProvider
+    public static Object[][] provideFinalTestResult() {
+        return new Object[][]{
+                {"*** Stats: SuiteContexts:  2", "SuiteContext"},
+                {"*** Stats: TestContexts:   2", "TestContext"},
+                {"*** Stats: ClassContexts:  5", "ClassContext"},
+                {"*** Stats: MethodContexts: 44", "MethodContexts"},
+                {"*** Stats: Test Methods Count: 41 (31 relevant)", "Test methods"},
+                {"*** Stats: Failed: 7", "Failed tests"},
+                {"*** Stats: Retried: 10", "Retried tests"},
+                {"*** Stats: Expected Failed: 6", "Expected failed tests"},
+                {"*** Stats: Skipped: 6", "Skipped tests"},
+                {"*** Stats: Passed: 12 ⊃ Recovered: 4 ⊃ Repaired: 1", "Passed tests"}
+        };
+    }
+
+    @BeforeClass
+    public void verifyLogFileExists() {
+        Assert.assertTrue(LOG_4_J_FILE_READER.existsFile(), "Log file exists.");
+    }
+
+    @Test(dataProvider = "provideExpectedStatusPerMethod")
+    public void testT01_verifySimpleTestStatus(final String methodName, final Status expectedTestStatus) {
+        LOG_4_J_FILE_READER.assertTestStatusPerMethod(METHOD_END_WORKER_SEARCH_TERM, methodName, expectedTestStatus);
+    }
+
+    @Test(dataProvider = "provideTestMethodsRetried")
+    public void testT02_verifyRetryStatus(final String methodName, final Status finalStatus) {
+
+        final String failedStatus = Status.FAILED.title;
+        final String finalStatusTitle = finalStatus.title;
+
+        // check basic test status in log
+        final List<String> methodEndEntries = LOG_4_J_FILE_READER.filterLogForTestMethod(METHOD_END_WORKER_SEARCH_TERM, methodName);
+
+        Assert.assertEquals(methodEndEntries.size(), 2,
+                "correct amount of test status entries found.");
```

Comment:
I cannot help but notice that all but the last message in this method claim that something be the case, which according to the assertion clearly is not. It might be better to invert the statements.

Reply:
Assertion messages should always decsribe the expected result and let the the tests readable.

Reply:
Like "Error: No error"? ... Okay.
In this case it looks like the last message seems to diverge: "entry '%s' has information on abortion of retry due to exceeded limit" reads to me like a description _of the fault_, not a description _of the expectation_ (but mayhap I just do not understand the check).